### DISCUSSION TITLE
Why not publish STIG Benchmark InSpec Profiles quarterly?

### DISCUSSION THREAD

**Discussion by salderma** [mitre/saf#1141] [Q&A] [answered]
Hi all,

Just discovered this tool. However, it does not seem to run well at all for me, I'm not sure if the documentation lacking or I'm hitting bugs. I'm running Ubuntu 20.04, have pulled various versions of the docker image and have attempted to work with the npm version. Neither have produced a successful inspec profile output for a RHEL8 benchmark.

But in my mind, if these inspec profiles are just text, the benchmarks are just text, and MITRE has this software deployed in a functioning manor, why not also generate and publish the InSpec profiles for the most/all OS Benchmarks as part of this project or a companion project?

P.S. It looks like a large number of benchmarks have been included in the repository for the purposes of CI testing, why not keep those benchmarks updated and publish the generated InSpec profiles?

**Comment by Amndeep7** [comment, marked as answer]
Hello @salderma, apologies for the delay in response.

The SAF CLI does not do any scanning in and of itself.  What it can do is generate an InSpec profile stub given a given STIG that you would then need to fill out.  We also have the 'delta' capability that could be used to prepopulate as many implementations as possible for controls that are similar between the two versions of a given STIG or even between two similar STIGs (ex. RHEL vs Ubuntu).  However, you would still need to go in afterwards and finish the implementation.

Once you have the InSpec profile, you would then need to use InSpec or CINC to run it against your target, at which point you could use various downstream capabilities provided by the SAF CLI and/or upload the results into Heimdall for visualization and data retention purposes.

W/r to why we don't have a process in place for keeping all InSpec profiles updated as per the latest versions of the benchmarks, due to the small size of our team and our funding model as an FFRDC, we are only able to prioritize profiles that our government sponsors are interested in having updated versions of.  We have done our best to provide tooling that will accelerate the pace at which the updates can happen once there is interest.  In the remainder of the time, we are reliant on the occasional free cycle or contributions from the community.
